Jaclyn taroni/2023 02 19 edits (#242)
jaybee84/ml-in-rd: content/05.model-complexity.md

Thus, applying ML to rare disease data without addressing the aforementioned shortcomings may lead to models that have low reproducibility or are hard to interpret.

Class imbalance in datasets can be addressed using decision tree-based ensemble learning methods (e.g., random forests). [@doi:10.1007/s11634-019-00354-x] (Figure[@fig:3]a)
Random forests use resampling (with replacement) based techniques to form a consensus about the important predictive features identified by the decision trees (e.g., Box 1c). [@https://doi.org/10.1023/A:1010933404324; @doi:10.1186/1472-6947-13-134]

> Additional approaches like combining random forests with resampling without replacement can generate confidence intervals for the model predictions by iteratively exposing the models to incomplete datasets, mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].

Reviewer: ```suggestion
Additional approaches like combining random forests with resampling without replacement can generate confidence intervals for the model predictions (e.g. for applications like Box 1d) by iteratively exposing the models to incomplete datasets, mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].
```

Author: ```suggestion
Additional approaches like combining random forests with resampling without replacement can generate confidence intervals for the model predictions (e.g., for applications like Box 1d) by iteratively exposing the models to incomplete datasets, mimicking real world cases where most rare disease datasets are incomplete [@doi:10.3390/genes11020226].
```

For example, LASSO has been used as a feature selection method for amyotrophic lateral sclerosis (ALS) gene expression data. [@doi:10.1186/s10020-023-00603-y]
In this example, applying LASSO regularization reduced the number of genes included as features in the a machine learning model designed to classify brain tissue regions from ALS patients. 
In the context of rare immune cell signature discovery, variations of elastic-net regression were found to outperform other regression approaches [@doi:10.1016/j.compbiomed.2015.10.008; @doi:10.1186/s12859-019-2994-z]. 

> Thus, regularization methods like LASSO or elastic-net are beneficial in ML with rare observations, and are worth exploring in the context of rare diseases.[@doi:10.1371/journal.pgen.1004754] <!--TODO: replace this reference-->

Reviewer: ```suggestion
Thus, regularization methods like LASSO or elastic-net are beneficial in ML with rare observations, and are worth exploring in the context of rare diseases.[@doi:10.1186/s10020-023-00603-y] 
```
